### PR TITLE
File renamings and code cleanup

### PR DESCRIPTION
**Pull request by dsyme**
This updates http://visualfsharp.codeplex.com/SourceControl/network/forks/dsyme/cleanup?branch=master-cleanup-1 to integrate into fsharp4.

This is a pass of cleanup suitable for the "master" branch that renames some files to get rid of most cryptic abbreviations like "csolve", and does some other code cleanup and documentation too.

 - commenting in "detuple.fs"
  - removing ExtensibleDumper.fs which is an old adhoc debug mechanism barely used in the codebase
  - removing some dead code in fsc.fs (some code was already duplicated in fscmain.fs too!)
  - renaming check.{fs,fsi} --> PostInferenceChecks.{fs,fsi}
  - renaming tc.{fs,fsi} --> TypeChecker.{fs,fsi}
  - renaming opt.{fs,fsi} --> Optimizer.{fs,fsi}
  - renaming est.{fs,fsi} --> ExtensionTyping.{fs,fsi}
  - renaming build.{fs,fsi} --> CompileOps.{fs,fsi}
  - renaming fscopts.{fs,fsi} --> CompileOptions.{fs,fsi}
  - moving the option parser to CompileOptions.fs (where it belongs!)
  - marking some record types as RequireQualifiedAccess (to give better errors when editing the compiler)
  - removed a whole bunch of semicolons
  - removed some old debugging output (verboseStamps etc.)

### CODE REVIEW

**Comment by dsyme** [comment]
Personally I think we should integrate this sooner rather than later.  We regularly integrate along the path Microsoft/visualfsharp --> fsharp/fsharp --> fsharp/FSharp.Compiler.Service for master and fsharp4 branches.  The cleanup can be relatively quickly integrated across all three repos and we can move on in a new, cleaner world.

**Comment by forki** [comment]
Oh I just learned to love the cool names like tc.fs. 

:+1:

File: src/fsharp/AugmentWithHashCompare.fs [hunk 1]
```diff
@@ -395,7 +396,7 @@ let mkUnionCompareWithComparer g tcref (tycon:Tycon) (_thisv,thise) (thatv,thate
     let expr = 
         let mbuilder = new MatchBuilder(NoSequencePointAtInvisibleBinding,m ) 
         let mkCase ucase =
-            let cref = mkNestedUnionCaseRef tcref ucase 
+            let cref = tcref.NestedUnionCaseRef ucase
```
Comment (forki) [inline src/fsharp/AugmentWithHashCompare.fs]: what was the `mk`? Shouldn't it be something like `tcref.MakeNestedUnionCaseRef`?

Comment (dsyme) [inline src/fsharp/AugmentWithHashCompare.fs]: Changed to Make*

File: src/fsharp/CompileOps.fs [hunk 2]
```diff
@@ -3389,38 +3378,30 @@ let PickleToResource file g scope rname p x =
...
-    PickleToResource file tcGlobals ccu (FSharpSignatureDataResourceName+"."+ccu.AssemblyName) pickleModuleInfo 
+    PickleToResource file tcGlobals ccu (FSharpSignatureDataResourceName+"."+ccu.AssemblyName) pickleCcuInfo 
         { mspec=mspec; 
           compileTimeWorkingDir=tcConfig.implicitIncludeDir;
           usesQuotations = ccu.UsesFSharp20PlusQuotations }
 #endif // NO_COMPILER_BACKEND
 
 let GetOptimizationData (file, ilScopeRef, ilModule, byteReader) = 
-    unpickleObjWithDanglingCcus file ilScopeRef ilModule Opt.u_LazyModuleInfo (byteReader())
+    unpickleObjWithDanglingCcus file ilScopeRef ilModule Optimizer.u_CcuOptimizationInfo (byteReader())
```
Comment (forki) [inline src/fsharp/CompileOps.fs]: is this the same thing?

Comment (dsyme) [inline src/fsharp/CompileOps.fs]: Yes

File: src/fsharp/vs/service.fs [hunk 3]
```diff
@@ -743,7 +743,7 @@ type TypeCheckInfo
     let GetClassOrRecordFieldsEnvironmentLookupResolutions(line,colAtEndOfNamesAndResidue, plid, (_residue : string option)) = 
         let cursorPos = Pos.fromVS line colAtEndOfNamesAndResidue
         let (nenv, ad),m = GetBestEnvForPos cursorPos
-        let items = Nameres.ResolvePartialLongIdentToClassOrRecdFields ncenv nenv m ad plid false
+        let items = NameResolution.ResolvePartialLongIdentToClassOrRecdFields ncenv nenv m ad plid false
```
Comment (forki) [inline src/fsharp/vs/service.fs]: Recd could be renamed to Record. It's only 2 chars more in a very long identifier

File: src/fsharp/vs/ServiceLexing.fs [hunk 4]
```diff
@@ -646,7 +646,7 @@ type internal LineTokenizer(text:string,
                   | RQUOTE_DOT (s,raw) -> 
                       delayToken(DOT, rightc, rightc)
                       false, (RQUOTE (s,raw), leftc, rightc - 1)
-                  | INFIX_COMPARE_OP (Lexfilter.TyparsCloseOp(greaters,afterOp) as opstr) -> 
+                  | INFIX_COMPARE_OP (LexFilter.TyparsCloseOp(greaters,afterOp) as opstr) ->
```
Comment (forki) [inline src/fsharp/vs/ServiceLexing.fs]: Maybe TyparsCloseOp could also be renamed

File: src/fsharp/tast.fs [hunk 5]
```diff
@@ -3109,7 +3398,7 @@ and CcuResolutionResult =
     | UnresolvedCcu of string
 
 /// Represents the information saved in the assembly signature data resource for an F# assembly
-and PickledModuleInfo =
+and PickledCcuInfo =
```
Comment (forki) [inline src/fsharp/tast.fs]: What's Ccu?

Comment (dsyme) [inline src/fsharp/tast.fs]: Compilation unit, see tast.fs

File: src/fsharp/vs/service.fs (hunk 3 above)
Comment (dsyme) [inline src/fsharp/vs/service.fs]: "Recd" is used pretty extensively in the codebase so it's consistent.  It's also kind of useful to distinguish it from the verb "Record".

File: src/fsharp/tast.fs (hunk 5 above)
Comment (forki) [inline src/fsharp/tast.fs]: mhm. I assume this is already established, but maybe something like CU would fit better. I wonnder where the second c comes from.

Comment (latkin) [inline src/fsharp/tast.fs]: Full acronym is "code compilation unit" IIRC

**Comment by latkin** [comment]
We were reluctant to take the original PR as it came in at a time when there were a number of other PRs pending which would have been a pain to merge.  That's no longer the case so this should be easier to accept now.  Any chance you are going to migrate ["cleanup referenceresolution.fs/fsi"](http://visualfsharp.codeplex.com/SourceControl/network/forks/dsyme/cleanup/contribution/7671) too?

**Comment by dsyme** [comment]
AppVeyor just failed with this, which looks disturbing, as though there may still be a race condition, see ``RaceBetweenCancellationHandlerAndDisposingHandlerRegistration``.  I'm pretty certain it's unrelated to this PR but will check.

```
 1) Test Failure : FSharp.Core.Unittests.FSharp_Core.Microsoft_FSharp_Control.AsyncModule.OnCancel.RaceBetweenCancellationHandlerAndDisposingHandlerRegistration
      Expected: True
   But was:  False

```

**Comment by dsyme** [comment]
@latkin - Yes, I'll move that other one too, thanks for reminding me

**Comment by KevinRansom** [comment]
These large cleanup PR's suck when reviewed using IE, it slows down to a crawl.

I like the file renaming and field renaming, the style is more similar to .Net style.  

Some of this cleanup looks like whitespace reformatting.  Is that for consistency? Or is it because you prefer this style?  Or perhaps an artifact of a reformat tool that you ran?  My preference is to avoid large scale whitespace refactoring and personal preference refactoring?

I believe Roslyn actually runs a tool that slams all code into a particular "authorized format" (and every time I hear about it, is because of some bug or other). If we wanted to codify a set of coding guidelines, we might want to decide them and publish them.  Although we do have quite a lot of other work coming up.

I'm okay with accepting this PR, but I would prefer to limit the personal preference refactoring to an agreed style moving forward.  What do you think?

Kevin

**Comment by forki** [comment]
Works well in Chrome.

**Comment by enricosada** [comment]
@dsyme i think you just lost some rename opportunity in git, like this you lose file history (or harder to follow).

for example in 31777c7fbf0f8e59fd8ef8967b618ecf76aa2b7b 
DELETED [src/fsharp/tc.fsi](https://github.com/dsyme/visualfsharp/commit/31777c7fbf0f8e59fd8ef8967b618ecf76aa2b7b#diff-28ee7226353c2b3c5b32425da4816f62) 
ADDED [src/fsharp/TypeChecker.fsi](https://github.com/dsyme/visualfsharp/commit/31777c7fbf0f8e59fd8ef8967b618ecf76aa2b7b#diff-ccea3561213c729c49403ff3d03d8fb8)

no rename here

maybe easier to split work in two phase (or use a more aggressive rename detection):
-  rename files + change fsproj
-  refactor content

**Comment by enricosada** [comment]
btw good work on cleanup!! :smile:

**Comment by dsyme** [comment]
@enricosada Do you know if I can get more aggressive rename detection applied by any technique? e.g by resubmitting the PR, or squashing etc? Thanks

**Comment by dsyme** [comment]
@KevinRansom All the changes are in some sense moving towards a more normalized style across files or across the compiler codebase as a whole -  consistency is what counts here. There are still plenty of inconsistencies but I think we can work on that bit by bit.

I'll aim to write up the conventions used as a set of guidelines, and perhaps also a set of settings for VFPT code formatting (which I haven't used here though it would make a marvellous stress test for the code formatter :) )

btw yes, use Chrome to view this PR :)

File: src/fsharp/tast.fs (hunk 5 above)
Comment (enricosada) [inline src/fsharp/tast.fs]: Maybe a glossary with all the acronyms can help. 
Easy doc and useful ( search  text `ccu` in repo -> glossary.md )

**Comment by enricosada** [comment]
@dsyme so, git does not track rename in commit, it does check for rename in diff or merge command. 
So you dont lose history (my bad) but is difficult to merge usually or see diff (you need to increase threshold when -follow-rename). 

So:
- the user increase the threshold of rename-detection in diff/merge (command line args) when needed
- you can help git splitting rename+modify. see [splitting commits with interactive rebase](http://git-scm.com/book/en/v2/Git-Tools-Rewriting-History#Splitting-a-Commit)

if you need help, ping me, i use rebase alot (i like to cleanup and split before pull request).

I think easier workflow is 

- git rebase -i
  - fixup last commits (the fix build), less commit to move
- git rebase -i
  - edit 31777c7
  - commit all good renames
  - commit deleted and the new rename (= deleted)
  - commit modify
  - rebase --continue

**Comment by KevinRansom** [comment]
Thank you for taking the time to take care of this work, we will take care of this at the start of the next cycle.

Moving this to V.Next per: #371

**Comment by dsyme** [comment]
I've updated this with a merge against the latest head of fsharp4. Let me know if you'd like me to send it to another branch.  It would be great to get these clean-ups in :)

**Comment by latkin** [comment]
Almost ready to apply this. One issue found by tests: https://github.com/dsyme/visualfsharp/commit/20cb9b0bf32d01cce556e37edee1ab075051c4a5#commitcomment-12477580

**Comment by dsyme** [comment]
OK, it looks like one of those cases should have been deleted but not both.  Have fixed that now with https://github.com/Microsoft/visualfsharp/commit/5e692365e8fb01f279aa7836dbd3e4d6e96fff0c

**Comment by latkin** [comment]
Applied to the `OOB` branch.

**Comment by dsyme** [comment]
Yay!

**Comment by forki** [comment]
So should we rebase our PRs on this? Would probably make things easier for
@latkin.

**Pull request closed without merging**